src/Castle.Facilities.AspNetCore/WindsorRegistrationExtensions.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright 2004-2018 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Facilities.AspNetCore
+{
+	using System;
+	using System.Linq;
+
+	using Castle.Core;
+	using Castle.Facilities.AspNetCore.Resolvers;
+	using Castle.Facilities.AspNetCore.Activators;
+	using Castle.Facilities.AspNetCore.Contributors;
+	using Castle.MicroKernel.Lifestyle;
+	using Castle.MicroKernel.Registration;
+	using Castle.MicroKernel.Resolvers.SpecializedResolvers;
+	using Castle.MicroKernel;
+	using Castle.Windsor;
+
+	using Microsoft.AspNetCore.Builder;
+	using Microsoft.AspNetCore.Http;
+	using Microsoft.AspNetCore.Mvc;
+	using Microsoft.AspNetCore.Razor.TagHelpers;
+	using Microsoft.Extensions.DependencyInjection;
+
+	public static class WindsorRegistrationExtensions
+	{
+		private static bool crossWiresIntoWasCalledBeforeRegistration = false;
```

Comment:
Doesn't this have to be called for each container, so this being static wouldn't work for a second container in the process?

Reply:
Awesome, I need a way of tacking a reference to the IServiceCollection on the container. I was hunting through the internals of the IKernel but then I just got grumpy and tired. 

Is there anyway this can be done? That way we don’t smash container life cycles.

Also, there can be only 1 service provider when it comes to the hosting runtime, but I don’t see why we can’t have multiple Windsor containers. 

You OK with me adding a general purpose key/value store to the kernel to solve this?

Reply:
> You OK with me adding a general purpose key/value store to the kernel to solve this?

Not really. This is why most facilities will have a facility instance which can hold whatever extra variables it needs and set up the facility in the container, I don't think we want to pollute the container with a generic KV store when we've got a pattern that works well.

I guess if you didn't want that you could have another component model contributor for this facility to check the configuration when you register, and check `kernel.ComponentModelBuilder.Contributors.Any(c => c.GetType() == typeof(CrossWiringComponentModelContributor))`, however you'd then be better off just always installing the `CrossWiringComponentModelContributor` and allowing `services` to be null, and `CrossWiringComponentModelContributor` will throw when you call `Register` not when you call `CrossWired` since it doesn't yet know which container the `ComponentRegistration` is going to be registered in. Even with a KV store on the container you don't know the container until `Register` is called, so this has to change.

Reply:
Ok in this case we should try housing the ServiceCollection reference in a facility first. I am almost 100% certain this will work. 

We can use the contributor as a fallback position if we don’t like the way it looks. 

Thanks for explaining.

Reply:
@generik0 

I will submit a review. Can you give it one last look. I will highlight the latest changes.